tests/ui_tests/reporting/testreport.py

```diff
@@ -213,12 +214,81 @@ def all_unique_screens() -> Path:
     return html.write(TESTREPORT_PATH, doc, ALL_UNIQUE_SCREENS)
 
 
-def generate_reports() -> None:
+def screen_text_report() -> None:
+    """Generate a report with text representation of all screens."""
+    recent_tests = list(TestResult.recent_tests())
+
+    with open(SCREEN_TEXT_FILE, "w") as f2:
+        for test in recent_tests:
+            if not test.test.screen_text_file.exists():
+                continue
+            f2.write(f"\n{test.test.id}\n")
+            with open(test.test.screen_text_file, "r") as f:
+                for line in f.readlines():
+                    f2.write(f"\t{line}")
+
+    # TODO: decide which is better - .txt or .html (maybe both?)
+
+    doc = dominate.document(title="Screen text report")
+    with doc:
+        for test in recent_tests:
+            if not test.test.screen_text_file.exists():
+                continue
+            with a(href=f"{ALL_SCREENS}#{test.test.id}"):
+                h2(test.test.id)
+            with open(test.test.screen_text_file, "r") as f:
+                for line in f.readlines():
+                    p(line)
+
+    html.write(TESTREPORT_PATH, doc, "screen_text.html")
+
+
+def differing_screens() -> None:
+    """Creating an HTML page showing all the unique screens that got changed."""
+    unique_diffs: set[tuple[str | None, str | None]] = set()
+
+    def already_included(left: str | None, right: str | None) -> bool:
+        return (left, right) in unique_diffs
+
+    def include(left: str | None, right: str | None) -> None:
+        unique_diffs.add((left, right))
+
+    recent_tests = list(TestResult.recent_tests())
+
+    model = recent_tests[0].test.model if recent_tests else None
+    doc = document(title="Differing screens", model=model)
+    with doc:
+        with table(border=1, width=600):
+            with tr():
+                th("Expected")
+                th("Actual")
+                th("Testcase (link)")
+            for test in recent_tests:
+                _, recorded_hashes = test.test.recorded_screens
+                _, actual_hashes = test.test.actual_screens
+
+                for recorded, actual in screens_diff(recorded_hashes, actual_hashes):
+                    if recorded != actual and not already_included(recorded, actual):
+                        include(recorded, actual)
+                        with tr(bgcolor="red"):
+                            html.image_column(recorded, TESTREPORT_PATH)
+                            html.image_column(actual, TESTREPORT_PATH)
+                            with td():
+                                with a(href=f"failed/{test.test.id}.html"):
```

Comment:
tiny nit but maybe use `result.test.id` for consistency with elsewhere
(yes I realize `TestResult.recent_tests()` should have been called `recent_results()` or just `recents()`)